src/source/reporting.rst

```diff
@@ -0,0 +1,285 @@
+``am:sessionEngagementCTR``        Double     Session    Engagement click-through rate, in session scope. Calculation: ``(am:sessionsWithWebsiteClicks + am:emails) / am:sessionsWithClicks``
+================================  =========  =========  ============================================================
+
+Scope
+######################################################
+A scope for a metric defines the level at which that metric is defined — hit, session, or user. For example, ``am:clicks`` and ``am:impressions`` are **hit** level metrics, since they occur in a hit, whereas ``am:sessionsWithClicks`` and ``am:sessionsWithImpressions`` are **session** level metrics since there is a single value for these per session. Conceptually, **user** is the highest level scope and **hit** is the lowest level scope. We plan to extend support for **user**-level scoped metrics if there is demand for it.
+
+
+Time Ranges
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _time-ranges:
+
+This field is used to specify one ore more distinct time periods to fetch data for, across which all other query parameters will be the same.
+Supplying a single range is sufficient for most needs, but the flexibility is provided to ask for multiple at the same time.
+
+.. _time-range:
+
+Each time range clause is defined as follows:
+
+.. code:: javascript
+
+    {
+        "period":   <string>
+        "from":     <string>
+        "to":       <string>
+    }
+
+There are several *predefined* period values at your disposal: ``today``, ``yesterday``, ``thisWeek``, ``lastWeek``, ``thisMonth``, ``lastMonth``, ``thisYear``, and ``lastYear``, with their obvious meanings.
+If you use a predefined period for a particular time range, the *from* and *to* fields of that time range will be ignored.
+
+To use a *custom* period, you need to provide the start (*from*) and end (*to*) date (**both inclusive**) in the following format: ``YYYY-MM-DD``.
+The end date cannot be before the start one. For example:
+
+.. code:: javascript
+
+    "timeRanges": [
+        {
+                "period":   "custom",
+                "from":     "2017-01-01",
+                "to":       "2017-10-22"
+        }
+    ]
+
+
+Example with multiple time ranges, mixed custom and predefined:
+
+.. code:: javascript
+
+    "timeRanges": [
+        {
+            "period":   "thisWeek"
+        },
+        {
+            "period":   "custom",
+            "from":     "2017-01-08",
+            "to":       "2017-01-15"
+        }
+    ]
+
+
+Time Aggregation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _time-aggregation:
+
+The ``aggregate`` parameter is used in combination with the ``am:date`` dimension, and is **mandatory** in such case. It specifies the granularity/resolution of the time according to which the metrics data will be grouped in buckets. Possible values are: ``daily``, ``weekly``, ``quarterly``, ``montly``, ``yearly``.
```

Comment:
which one is mandatory? date or aggregate?

Reply:
:) `aggregate`... will try to reword